cmd/waku/relay.go

```diff
@@ -0,0 +1,147 @@
+	"time"
+
+	"github.com/waku-org/go-waku/logging"
+	"github.com/waku-org/go-waku/waku/v2/node"
+	wprotocol "github.com/waku-org/go-waku/waku/v2/protocol"
+	"github.com/waku-org/go-waku/waku/v2/protocol/relay"
+	"github.com/waku-org/go-waku/waku/v2/rendezvous"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+)
+
+var fwdMetaTag = []byte{102, 119, 100} //"fwd"
+
+func handleRelayTopics(ctx context.Context, wg *sync.WaitGroup, wakuNode *node.WakuNode, pubSubTopicMap map[string][]string) error {
+	for nodeTopic, cTopics := range pubSubTopicMap {
+		nodeTopic := nodeTopic
+		_, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(nodeTopic, cTopics...), relay.WithoutConsumer())
+		if err != nil {
+			return err
+		}
+
+		if len(options.Rendezvous.Nodes) != 0 {
+			// Register the node in rendezvous point
+			iter := rendezvous.NewRendezvousPointIterator(options.Rendezvous.Nodes)
+
+			wg.Add(1)
+			go func(nodeTopic string) {
+				t := time.NewTicker(rendezvous.RegisterDefaultTTL)
+				defer t.Stop()
+				defer wg.Done()
+
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case <-t.C:
+						// Register in rendezvous points periodically
+						wakuNode.Rendezvous().RegisterWithNamespace(ctx, nodeTopic, iter.RendezvousPoints())
+					}
+				}
+			}(nodeTopic)
+
+			wg.Add(1)
+			go func(nodeTopic string) {
+				defer wg.Done()
+				desiredOutDegree := wakuNode.Relay().Params().D
+				t := time.NewTicker(7 * time.Second)
+				defer t.Stop()
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case <-t.C:
+						peerCnt := len(wakuNode.Relay().PubSub().ListPeers(nodeTopic))
+						peersToFind := desiredOutDegree - peerCnt
+						if peersToFind <= 0 {
+							continue
+						}
+
+						rp := <-iter.Next(ctx)
+						if rp == nil {
+							continue
+						}
+						ctx, cancel := context.WithTimeout(ctx, 7*time.Second)
+						wakuNode.Rendezvous().DiscoverWithNamespace(ctx, nodeTopic, rp, peersToFind)
+						cancel()
+					}
+				}
+			}(nodeTopic)
+
+		}
+	}
+
+	// Bridge topics
+	bridgedTopics := make(map[string]map[string]struct{})
+	bridgedTopicsSet := make(map[string]struct{})
+	for _, topics := range options.Relay.BridgeTopics {
+		_, ok := bridgedTopics[topics.FromTopic]
+		if !ok {
+			bridgedTopics[topics.FromTopic] = make(map[string]struct{})
+		}
+
+		bridgedTopics[topics.FromTopic][topics.ToTopic] = struct{}{}
+		bridgedTopicsSet[topics.FromTopic] = struct{}{}
+		bridgedTopicsSet[topics.ToTopic] = struct{}{}
+	}
+
+	// Make sure all topics are subscribed
+	for _, topic := range maps.Keys(bridgedTopicsSet) {
+		if !wakuNode.Relay().IsSubscribed(topic) {
+			_, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(topic), relay.WithoutConsumer())
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for fromTopic, toTopics := range bridgedTopics {
+		subscriptions, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(fromTopic))
```

Comment:
Rather than publishing all messages on defaultTopic to the new shard. 
Wouldn't it be better to filter using contentTopics using by status app? 
DefaultTopic may be used by other users as well, and we would be flooding the shard 32 with unnecessary messages if we don't filter by contentTopics.

Reply:
The problem is that a different content topic is generated per each channel you have joined so it's not possible to have a list of content topics to filter.